TestHTMLFeatureInfoDocs.test_combine failure (IndexError: list index out of range)

The Debian package build fails after lxml was rebuilt with libxml2 2.14:
```python
=================================== FAILURES ===================================
_____________________ TestHTMLFeatureInfoDocs.test_combine _____________________

self = <mapproxy.test.unit.test_featureinfo.TestHTMLFeatureInfoDocs object at 0x7fbd5cb8a3f0>

    def test_combine(self):
        docs = [
            HTMLFeatureInfoDoc(b"<html><head><title>Hello<body><p>baz</p><p>baz2"),
            HTMLFeatureInfoDoc(b"<p>foo</p>"),
            HTMLFeatureInfoDoc(b"<body><p>bar</p></body>"),
        ]
>       result = HTMLFeatureInfoDoc.combine(docs)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

mapproxy/test/unit/test_featureinfo.py:167: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
mapproxy/featureinfo.py:125: in combine
    result_tree.body.extend(body)
    ^^^^^^^^^^^^^^^^
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <Element html at 0x7fbd5c9197c0>

    @property
    def body(self):
        """
        Return the <body> element.  Can be called from a child element
        to get the document's head.
        """
>       return self.xpath('//body|//x:body', namespaces={'x':XHTML_NAMESPACE})[0]
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
E       IndexError: list index out of range

/usr/lib/python3/dist-packages/lxml/html/__init__.py:266: IndexError
__________________ TestHTMLFeatureInfoDocs.test_combine_parts __________________

self = <mapproxy.test.unit.test_featureinfo.TestHTMLFeatureInfoDocs object at 0x7fbd5cb8afd0>

    def test_combine_parts(self):
        docs = [
            HTMLFeatureInfoDoc("<p>foo</p>"),
            HTMLFeatureInfoDoc("<body><p>bar</p></body>"),
            HTMLFeatureInfoDoc("<html><head><title>Hello<body><p>baz</p><p>baz2"),
        ]
        result = HTMLFeatureInfoDoc.combine(docs)
    
>       assert (
            "<body><p>foo</p><p>bar</p><p>baz</p><p>baz2</p></body>"
            in result.as_string()
        )
E       AssertionError: assert '<body><p>foo</p><p>bar</p><p>baz</p><p>baz2</p></body>' in '<html><body><p>foo</p><p>bar</p><head><title>Hello&lt;body&gt;&lt;p&gt;baz&lt;/p&gt;&lt;p&gt;baz2</title></head></body></html>'
E        +  where '<html><body><p>foo</p><p>bar</p><head><title>Hello&lt;body&gt;&lt;p&gt;baz&lt;/p&gt;&lt;p&gt;baz2</title></head></body></html>' = as_string()
E        +    where as_string = <mapproxy.featureinfo.HTMLFeatureInfoDoc object at 0x7fbd5c6a8d50>.as_string

mapproxy/test/unit/test_featureinfo.py:183: AssertionError
```

--- a/mapproxy/test/system/test_wms.py
+++ b/mapproxy/test/system/test_wms.py
@@ -753,7 +753,7 @@ class TestWMS111(SysTest):
             self.common_fi_req.params["info_format"] = "text/html"
             resp = app.get(self.common_fi_req)
             assert resp.content_type == "text/html"
-            assert resp.body == b"<html><body><p>info</p></body></html>"
+            assert resp.body == b"<html><body>info</body></html>"
             assert resp.headers['Content-Type'] == 'text/html; charset=utf-8'
 
     def test_get_featureinfo_info_format_special_chars(self, app):
@@ -771,7 +771,7 @@ class TestWMS111(SysTest):
             self.common_fi_req.params["info_format"] = "text/html"
             resp = app.get(self.common_fi_req)
             assert resp.content_type == "text/html"
-            assert resp.body == encode(u"<html><body><p>äüß▼</p></body></html>")
+            assert resp.body == encode(u"<html><body>äüß▼</body></html>")
             assert resp.headers['Content-Type'] == 'text/html; charset=utf-8'
 
     def test_get_featureinfo_130(self, app):

--- a/mapproxy/test/unit/test_featureinfo.py
+++ b/mapproxy/test/unit/test_featureinfo.py
@@ -160,7 +160,7 @@ class TestHTMLFeatureInfoDocs(object):
 
     def test_combine(self):
         docs = [
-            HTMLFeatureInfoDoc(b"<html><head><title>Hello<body><p>baz</p><p>baz2"),
+            HTMLFeatureInfoDoc(b"<html><head><title>Hello</title></head><body><p>baz</p><p>baz2</body></html>"),
             HTMLFeatureInfoDoc(b"<p>foo</p>"),
             HTMLFeatureInfoDoc(b"<body><p>bar</p></body>"),
         ]
@@ -176,7 +176,7 @@ class TestHTMLFeatureInfoDocs(object):
         docs = [
             HTMLFeatureInfoDoc("<p>foo</p>"),
             HTMLFeatureInfoDoc("<body><p>bar</p></body>"),
-            HTMLFeatureInfoDoc("<html><head><title>Hello<body><p>baz</p><p>baz2"),
+            HTMLFeatureInfoDoc("<html><head><title>Hello</title></head><body><p>baz</p><p>baz2</body></html>"),
         ]
         result = HTMLFeatureInfoDoc.combine(docs)
 

--- a/requirements-tests.txt
+++ b/requirements-tests.txt
@@ -29,7 +29,7 @@ jsonpointer==2.4
 jsonschema==4.17;python_version<"3.10"
 jsonschema==4.20;python_version>="3.10"
 junit-xml==1.9
-lxml==5.3.0
+lxml==6.0.2
 MarkupSafe==1.1.1
 mock==5.1.0
 more-itertools==10.1.0

--- a/setup.py
+++ b/setup.py
@@ -12,7 +12,7 @@ install_requires = [
     'Pillow>=10;python_version=="3.11"',
     'Pillow>=10.1;python_version=="3.12"',
     'Pillow>=11;python_version=="3.13"',
-    'lxml>=4',
+    'lxml>=6',
     'shapely>=2'
 ]
 

